wallet/src/main/java/com/android/identity_credential/wallet/presentation/PresentationFlowActivity.kt

```diff
@@ -0,0 +1,294 @@
+import com.android.identity_credential.wallet.R
+import com.android.identity_credential.wallet.WalletApplication
+import com.android.identity_credential.wallet.presentation.PresentationFlowActivity.PromptUnsuccessfulException
+import com.android.identity_credential.wallet.ui.prompt.biometric.showBiometricPrompt
+import com.android.identity_credential.wallet.ui.prompt.consent.showConsentPrompt
+import com.android.identity_credential.wallet.ui.prompt.passphrase.showPassphrasePrompt
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlinx.datetime.Clock
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+
+/**
+ * A base Activity that provides functions for initiating an MDL Presentation flow and obtaining the
+ * response bytes to send to the requesting party.
+ *
+ * Subclassing Activities call function [showPresentationFlow] with the request bytes parsed to a
+ * [PresentationRequestData]. Response bytes are (eventually) generated after authentication prompts
+ * validate the user successfully, else an exception is thrown.
+ *
+ * Exceptions are thrown for numerous reasons such as,
+ * - [IllegalStateException] if credentials could not be found in the Document; authentication
+ * type was not specified when showing Biometric prompt; a new secure area has not been handled yet
+ * - [PromptUnsuccessfulException] if a prompt is unsuccessful when user cancels the prompt or is
+ * unable to authenticate.
+ */
+open class PresentationFlowActivity : FragmentActivity() {
+    companion object {
+        const val TAG = "PresentationFlowActivity"
+    }
+
+    // reference WalletApplication for obtaining dependencies
+    private val walletApp: WalletApplication by lazy {
+        application as WalletApplication
+    }
+
+    // private class dedicated to managing/providing Mdoc-related functions needed by this Activity
+    private val mdocHelper = MdocHelper()
+
+    /**
+     * Show the Presentation Flow starting with Consent prompt, followed by Biometric prompt
+     * and/or Passphrase prompt. If all the prompts are successful, return the response bytes.
+     * Otherwise throw the Exception that prevented the Presentation flow from generating the
+     * response bytes.
+     *
+     * @param presentationRequestData the request data produced after parsing request bytes
+     * @return the response bytes generated after all prompts succeed.
+     * @throws [IllegalStateException] for incorrect configurations or if cannot find credential.
+     * @throws [PromptUnsuccessfulException] if a prompt does not succeed because the user cancelled
+     * or is unable to authenticate.
+     */
+    suspend fun showPresentationFlow(presentationRequestData: PresentationRequestData): ByteArray {
+        val credential = presentationRequestData.document.getMdocCredential()
+            ?: throw IllegalStateException("Unable to find credential in Document ${presentationRequestData.document.name}")
```

Comment:
Do you really need to introduce a Document extension for this? I mean
```
val credential = document.findCredential(
  WalletApplication.CREDENTIAL_DOMAIN_MDOC,
  Clock.System.now()
) as MdocCredential? ?: throw IllegalStateException("No MdocCredential in document")
```
is a lot easier to read and does the same. The problem with document.getMdocCredential() is that your reader will first look in Document and then when they can't find the method they realize you declared an extension somewhere. Just makes the code hard to read.